server/sonar-server/src/test/java/org/sonar/server/qualitygate/QualityGatesTest.java

```diff
@@ -315,60 +300,6 @@ public void should_fail_delete_condition() {
   }
 
   @Test
-  public void should_associate_project() {
-    Long qGateId = QUALITY_GATE_ID;
-    Long projectId = 24L;
-    when(dao.selectById(dbSession, qGateId)).thenReturn(new QualityGateDto().setId(qGateId));
-    underTest.associateProject(qGateId, projectId);
-    verify(dao).selectById(dbSession, qGateId);
-    ArgumentCaptor<PropertyDto> propertyCaptor = ArgumentCaptor.forClass(PropertyDto.class);
-    verify(propertiesDao).saveProperty(propertyCaptor.capture());
-    PropertyDto property = propertyCaptor.getValue();
-    assertThat(property.getKey()).isEqualTo("sonar.qualitygate");
-    assertThat(property.getResourceId()).isEqualTo(projectId);
-    assertThat(property.getValue()).isEqualTo("42");
-  }
-
-  @Test
-  public void associate_project_with_project_admin_permission() {
-    userSessionRule.set(authorizedProjectAdminUserSession);
-
-    Long qGateId = QUALITY_GATE_ID;
-    Long projectId = 24L;
-    when(dao.selectById(dbSession, qGateId)).thenReturn(new QualityGateDto().setId(qGateId));
-    underTest.associateProject(qGateId, projectId);
-    verify(dao).selectById(dbSession, qGateId);
-    ArgumentCaptor<PropertyDto> propertyCaptor = ArgumentCaptor.forClass(PropertyDto.class);
-    verify(propertiesDao).saveProperty(propertyCaptor.capture());
-    PropertyDto property = propertyCaptor.getValue();
-    assertThat(property.getKey()).isEqualTo("sonar.qualitygate");
-    assertThat(property.getResourceId()).isEqualTo(projectId);
-    assertThat(property.getValue()).isEqualTo("42");
-  }
-
-  @Test
-  public void should_dissociate_project() {
```

Comment:
it is fully tested by DeselectActionTest. The method QualityGates#dissociateProject() should be moved to DeselectAction.